include/deal.II/base/tensor.h

```diff
@@ -2028,7 +2004,9 @@ template <int rank_1,
           int rank_2,
           int dim,
           typename Number,
-          typename OtherNumber>
+          typename OtherNumber,
+          typename = typename std::enable_if<rank_1 != 0>::type,
+          typename = typename std::enable_if<rank_2 != 0>::type>
```

Comment:
You can reduce this to a single `std::enable_if` if you just connected the conditions with `&&` :-) But I think that the right condition is actually `rank_1 + rank_2 >= 2`.